samples/opengl/depthMapTriangulation/image_triangulator.cpp

```diff
@@ -0,0 +1,949 @@
+                    if (y2 >= 0) dependency_graph(y, x).push_back(Point2D(y2, x2));
+                }
+                else  // false: right/left/center-row
+                {
+                    if (side_center_flag == true)  // true: right/left
+                    {
+                        x1 = x + d_l;
+                        x2 = x - d_l;
+                        y1 = y;
+                        y2 = y;
+
+                        // add the dependency_graph if they don't point outside of the whole quadtree
+                        if (x1 < RQT_side_len) dependency_graph(y, x).push_back(Point2D(y1, x1));
+                        if (x2 >= 0) dependency_graph(y, x).push_back(Point2D(y2, x2));
+                    }
+                    else  // false: center
+                    {
+                        if (diagonal_dependency_dir == true)  // true: left down and right up
+                        {
+                            x1 = x + d_l;
+                            x2 = x - d_l;
+                            y1 = y - d_l;
+                            y2 = y + d_l;
+
+                            // add the dependency_graph if they don't point outside of the whole quadtree
+                            if (x1 < RQT_side_len && y1 >= 0) dependency_graph(y, x).push_back(Point2D(y1, x1));
+                            if (x2 >= 0 && y2 < RQT_side_len) dependency_graph(y, x).push_back(Point2D(y2, x2));
+                        }
+                        else  // false: left up and right down
+                        {
+                            x1 = x + d_l;
+                            x2 = x - d_l;
+                            y1 = y + d_l;
+                            y2 = y - d_l;
+
+                            // add the dependency_graph if they don't point outside of the whole quadtree
+                            if (x1 < RQT_side_len && y1 < RQT_side_len)
+                                dependency_graph(y, x).push_back(Point2D(y1, x1));
+                            if (x2 >= 0 && y2 >= 0) dependency_graph(y, x).push_back(Point2D(y2, x2));
+                        }
+
+                        // diagonal dependency_graph are always alternating
+                        diagonal_dependency_dir = !diagonal_dependency_dir;
+                    }
+
+                    // flip the flag since the next loop is the other kind of column
+                    side_center_flag = !side_center_flag;
+                }
+            }
+
+            // flip the flag since the next loop is the other kind of row
+            row_flag = !row_flag;
+        }
+    }
+
+    // clear tmp dependency_graph from level 0
+    tmp_vec.clear();
+    dependency_graph(0, 0)                               = tmp_vec;
+    dependency_graph(RQT_side_len - 1, 0)                = tmp_vec;
+    dependency_graph(0, RQT_side_len - 1)                = tmp_vec;
+    dependency_graph(RQT_side_len - 1, RQT_side_len - 1) = tmp_vec;
```

Comment:
Writing `dependency_graph(0, 0).clear()` is probably easier to understand than assigning an empty named vector to it.